modules/ROOT/pages/clauses/match.adoc

```diff
@@ -490,3 +492,124 @@ The above query uses the xref:functions/aggregating.adoc#functions-collect[`coll
 
 For more information about how Cypher queries work, see xref:clauses/clause-composition.adoc[].
 
+[role=label--new-5.26]
+[[dynamic-match]]
+== MATCH nodes and relationships using dynamic node labels and relationship types
+
+Node labels and relationship types can be referenced dynamically in expressions, parameters, and variables when matching nodes and relationships.
+This allows for more flexible queries and mitigates the risk of Cypher injection.
+(For more information about Cypher injection, see link:https://neo4j.com/developer/kb/protecting-against-cypher-injection/[Neo4j Knowledge Base -> Protecting against Cypher injection]).
+
+.Syntax for matching node labels dynamically
+[source, syntax]
+----
+MATCH (n:$(<expr>))
+MATCH (n:$any(<expr>))
+MATCH (n:$all(<expr>))
+----
+
+[NOTE]
+`MATCH (n:$all(<expr>))` is functionally equivalent to `MATCH (n:$(<expr>))`.
+
+.Syntax for matching relationship types dynamically
+[source, syntax]
+----
+MATCH ()-[r:$(<expr>))]->()
+MATCH ()-[r:$any(<expr>)]->()
+MATCH ()-[r:$all(<expr>))]->()
+----
+
+The expression must evaluate to a `STRING NOT NULL | LIST<STRING NOT NULL> NOT NULL` value.
+If you use a `LIST<STRING>` with more than one item in a relationship pattern with dynamic relationship types, no results will be returned.
+This is because a relationship can only have exactly one type.
+
+.Match labels dynamically
+[source, cypher]
+----
+WITH ["Person", "Director"] AS labels
+MATCH (directors:$(labels))
+RETURN directors
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| directors
+
+| (:Person:Director {name: "Oliver Stone"})
+| (:Person:Director {name: "Rob Reiner"})
+
+1+d|Rows: 2
+|===
+
+.Match nodes dynamically using the `any()` function
+[source, cypher]
+----
+MATCH (n:$any(["Movie", "Actor"]))
+RETURN n AS nodes
+----
+
+[NOTE]
+The xref:functions/predicate.adoc#functions-any[`any()`] function matches nodes that have any of the specified labels.
```

Comment:
This sounds like the any in the match is the any() function, but it isn't. It's behaviour is modeled off of the any function, but it is not identical, so maybe clarify this part a little more